mqtt/mqtt-broker-tests-util/src/client.rs

```diff
@@ -240,27 +239,40 @@ where
         let (sub_sender, sub_receiver) = mpsc::unbounded_channel();
         let (conn_sender, conn_receiver) = mpsc::unbounded_channel();
 
-        let (termination_handle, tx) = oneshot::channel::<()>();
+        let (termination_handle, termination_receiver) = mpsc::channel::<()>(1);
 
         let event_loop_handle = tokio::spawn(async move {
-            let event_loop = async {
-                while let Some(event) = client.next().await {
-                    let event = event.expect("event expected");
-                    match event {
-                        Event::NewConnection { .. } => conn_sender
-                            .send(event)
-                            .expect("can't send an event to a conn channel"),
-                        Event::Publication(publication) => pub_sender
-                            .send(publication)
-                            .expect("can't send an event to a pub channel"),
-                        Event::SubscriptionUpdates(_) => sub_sender
-                            .send(event)
-                            .expect("can't send an event to a sub channel"),
-                    };
-                }
-            };
-            pin_mut!(event_loop);
-            select(event_loop, tx).await;
+            let mut termination_receiver = termination_receiver.fuse();
+            let mut client = client.fuse();
+
+            loop {
+                select! {
```

Comment:
Yeah, every time I try to use `select` macro it doesn't look nice and makes a lot of nesting